tracing-attributes/src/lib.rs

```diff
@@ -154,12 +156,9 @@ pub fn instrument(args: TokenStream, item: TokenStream) -> TokenStream {
     let param_names: Vec<Ident> = params
         .clone()
         .into_iter()
-        .filter_map(|param| match param {
-            FnArg::Typed(PatType { pat, .. }) => match *pat {
-                Pat::Ident(PatIdent { ident, .. }) => Some(ident),
-                _ => None,
-            },
-            _ => None,
+        .flat_map(|param| match param {
+            FnArg::Typed(PatType { pat, .. }) => param_names(*pat),
+            _ => Box::new(iter::empty()),
```

Comment:
I think we should probably match the [`FnArg::Receiver`](https://docs.rs/syn/1.0.5/syn/enum.FnArg.html#variant.Receiver) case here as well, so that we can start including `self` in the generated spans. We can add that in a follow-up PR though, it's not a blocker for this branch.

Reply:
That makes sense!  I thought briefly about that but decided to keep this PR focused.

Reply:
I went ahead and added it, since it was only a few lines plus test.